### PR TITLE
feat(guard): add fill_to_target to maintain n_samples with exclude mode

### LINKED ISSUE #38: `guard` の `mode: exclude` 使用時に `n_samples` で指定した行数が確保されない
`guard` 設定で `mode: exclude` を使用すると、制約違反があった行が削除されるため、最終的な出力行数が `generator.n_samples` で指定した数よりも少なくなります。

金融データの監査においては、指定されたサンプルサイズを維持することが望ましいケースが多いため、不足分をリサンプリングして補填するなどのオプション（例: `fill_to_target: true`）があると便利です。

### PR DESCRIPTION
**Pull request by takurot**
## Summary

Fixes #38 — when `guard.mode: exclude` removes violating rows, the final sample count falls short of `generator.n_samples`.

- Add `fill_to_target: bool = False` field to `GuardSettings` (config model)
- Add `fill_to_target` parameter to `GuardEngine.__init__` and propagate via factory
- `BaselineGenerator.sample()` loops iteratively (up to 10 attempts, 2× oversample per batch) when `guard.fill_to_target=True` and `mode=exclude`, truncating to exactly `n_samples` once collected
- When all rows are always rejected (extreme case), returns the empty DataFrame instead of hanging

## Example config usage

```yaml
guard:
  enabled: true
  mode: exclude
  fill_to_target: true   # ← new option
  params:
    non_negative:
      - columns: [amount]
```

## Tests

- `tests/unit/guard/test_guard_fill_to_target.py` — unit tests for `GuardEngine.fill_to_target` flag
- `tests/integration/test_guard_fill_to_target_integration.py` — integration tests via `BaselineGenerator`:
  - Without fill: exclude mode reduces count
  - With fill: exact `n_samples` rows returned
  - Low rejection rate: still produces exact count
  - Config-driven: `GuardSettings → create_guard_engine` propagates flag correctly
  - Non-exclude mode: fill has no effect (detect mode keeps all rows)
  - All-reject edge case: returns empty DataFrame safely

```
pytest tests/unit/guard/test_guard_fill_to_target.py tests/integration/test_guard_fill_to_target_integration.py
# 11 passed
pytest  # 147 passed, 2 skipped
```

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: sfdao/generator/baseline.py [hunk 1]
```diff
@@ -97,6 +101,46 @@ def sample(self, n_samples: int) -> pd.DataFrame:
 
         return df
 
+    def _sample_with_fill(self, rng: np.random.Generator, n_samples: int) -> pd.DataFrame:
+        """Sample with guard EXCLUDE mode, resampling until n_samples rows pass constraints."""
+        guard = self.guard  # guard is guaranteed non-None by the caller
+        if guard is None:
+            return pd.DataFrame(columns=self._column_order)
+        MAX_ATTEMPTS = 10
```
Comment (takurot) [inline sfdao/generator/baseline.py]: [blocking] With `fill_to_target=True`, this still under-fills badly once the rejection rate is high. For example, a rule that keeps only 1% of rows makes `sample(100)` return 20 rows here, because `MAX_ATTEMPTS=10` and `batch_size=max(needed*2, n_samples)` cap the total accepted rows long before the target is reached. Since the setting is exposed as “maintain `n_samples`”, silently returning a short frame makes the flag unreliable. We should either keep sampling until the target is reached (and fail explicitly if progress stalls), or surface the under-fill to the caller instead of returning fewer rows without notice.

File: sfdao/generator/baseline.py [hunk 2]
```diff
@@ -97,6 +101,46 @@ def sample(self, n_samples: int) -> pd.DataFrame:
...
+                batch_df, _ = self.scenario.apply(batch_df)
+
+            batch_df, _ = guard.apply(batch_df)
+
+            if not batch_df.empty:
+                accumulated.append(batch_df)
+                accumulated_count += len(batch_df)
+
+        if not accumulated:
+            return pd.DataFrame(columns=self._column_order)
```
Comment (takurot) [inline sfdao/generator/baseline.py]: [blocking] `pd.DataFrame(columns=self._column_order)` loses the fitted schema when every batch is rejected. In the existing non-fill path, `guard.apply()` preserves dtypes even for an empty result, but this branch turns numeric/datetime columns into `object`. That is a behavior regression for downstream evaluators that inspect dtypes. Please return an empty frame with the original column dtypes instead of rebuilding from column names only.

**Comment by takurot** [comment]
レビューで指摘した 2 点に対応しました。`BaselineGenerator._sample_with_fill()` は empty 時も schema を維持し、高除外率では固定 10 回打ち切りではなく進捗に応じて再サンプリングするようにしています。あわせて high-rejection と typed-empty の回帰テストを追加しました。push 後に CI を監視します。
